software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlSshDriver.java

```diff
@@ -328,9 +340,39 @@ private void initializeNewDatabase() {
                         " --command="+ createUserCommand),
                 sudoAsUser("postgres", getInstallDir() + "/bin/psql -p " + entity.getAttribute(PostgreSqlNode.POSTGRESQL_PORT) + 
                                 " --command="+ createDatabaseCommand),
+                createRolesAdditionalCommand,
                 callPgctl("stop", true))
                 .failOnNonZeroResultCode().execute();
     }
+
+    private String buildCreateRolesQuery() {
+        Map<String, Map> roles = entity.getConfig(PostgreSqlNode.ROLES);
+        String createRolesQuery = "\"";
+
+        for (Map.Entry role: roles.entrySet()) {
+            createRolesQuery = createRolesQuery.concat(String.format("CREATE ROLE %s", role.getKey()));
```

Comment:
Actually I'm not sure you're going to be able to use prepared statements with DDL like CREATE ROLE, at least not with Postgres.  (Worth a look to see whether it's supported anyway.)  At the very least, as Thomas says, you should be validating the input before using it in a statement.